mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -140,21 +140,27 @@ getArchAttributes(Operation *op, Type inputType) {
 }
 
 static FailureOr<rock::ConvOp>
-makeRockConv2D(ConversionPatternRewriter &rw, Operation *op, Value input,
-               Value filter, Value output, const DenseI64ArrayAttr &pad,
-               const DenseI64ArrayAttr &stride,
-               const DenseI64ArrayAttr &dilation, int64_t group) {
+makeRockConv(ConversionPatternRewriter &rw, Operation *op, Value input,
+             Value filter, Value output, const DenseI64ArrayAttr &pad,
+             const DenseI64ArrayAttr &stride, const DenseI64ArrayAttr &dilation,
+             int64_t group) {
   Location loc = op->getLoc();
 
-  SmallString<5> filterLayout("kyxc");
+  SmallString<7> filterLayout("kyxc");
   if (auto attr = op->getAttrOfType<StringAttr>("filter_layout"))
     filterLayout = attr.getValue();
-  SmallString<5> inputLayout("nhwc");
+  else if (filter.getType().template cast<ShapedType>().getRank() > 4)
```

Comment:
Really not "> 4" as much as "== 5";  ie, not general.  We also have mixed-mode layouts here, with x/y/h/w for 2-D and 0/1/2 for 3-D.  Some existing rocMLIR tests failed if I used "k01c" and I didn't want to spend the time to track it down.  Will need a cleanup ticket later.

Reply:
What tests, and failed how?